src/main/java/io/supertokens/webserver/api/bulkimport/BulkImportAPI.java

```diff
@@ -0,0 +1,202 @@
+        Integer limit = InputParser.getIntQueryParamOrThrowError(req, "limit", true);
+
+        if (limit != null) {
+            if (limit > BulkImport.GET_USERS_PAGINATION_LIMIT) {
+                throw new ServletException(
+                        new BadRequestException("Max limit allowed is " + BulkImport.GET_USERS_PAGINATION_LIMIT));
+            } else if (limit < 1) {
+                throw new ServletException(new BadRequestException("limit must a positive integer with min value 1"));
+            }
+        } else {
+            limit = BulkImport.GET_USERS_DEFAULT_LIMIT;
+        }
+
+        BULK_IMPORT_USER_STATUS status = null;
+        if (statusString != null) {
+            try {
+                status = BULK_IMPORT_USER_STATUS.valueOf(statusString);
+            } catch (IllegalArgumentException e) {
+                throw new ServletException(new BadRequestException("Invalid value for status. Pass one of NEW, PROCESSING, or FAILED!"));
+            }
+        }
+
+        AppIdentifier appIdentifier = null;
+        Storage storage = null;
+    
+        try {
+            appIdentifier = getAppIdentifier(req);
+            storage = enforcePublicTenantAndGetPublicTenantStorage(req);
+        } catch (TenantOrAppNotFoundException | BadPermissionException e) {
+            throw new ServletException(e);
+        }
+
+        try {
+            BulkImportUserPaginationContainer users = BulkImport.getUsers(appIdentifier, storage, limit, status, paginationToken);
+            JsonObject result = new JsonObject();
+            result.addProperty("status", "OK");
+
+            JsonArray usersJson = new JsonArray();
+            for (BulkImportUser user : users.users) {
+                usersJson.add(user.toJsonObject());
+            }
+            result.add("users", usersJson);
+
+            if (users.nextPaginationToken != null) {
+                result.addProperty("nextPaginationToken", users.nextPaginationToken);
+            }
+            super.sendJsonResponse(200, result, resp);
+        } catch (BulkImportUserPaginationToken.InvalidTokenException e) {
+            Logging.debug(main, null, Utils.exceptionStacktraceToString(e));
+            throw new ServletException(new BadRequestException("invalid pagination token"));
+        } catch (StorageQueryException e) {
+            throw new ServletException(e);
+        }
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        // API is app specific
+
+        if (StorageLayer.isInMemDb(main)) {
+            throw new ServletException(new BadRequestException("This API is not supported in the in-memory database."));
+        }
+
+        JsonObject input = InputParser.parseJsonObjectOrThrowError(req);
+        JsonArray users = InputParser.parseArrayOrThrowError(input, "users", false);
+
+        if (users.size() <= 0 || users.size() > BulkImport.MAX_USERS_TO_ADD) {
```

Comment:
in case it's 0, you just return 200 status code with correct json output.

Reply:
I think its better to return an error because calling this API with 0 users is not a valid case.

Reply:
it makes it harder for people to call the API. Cause if they have a loop in which they are calling the API, they will be forced to check if the input is empty before calling the API (as opposed to being flexible enough to allow them to check it after and then exit the loop)

Reply:
That makes sense. I have updated it.